src/dbus.cpp

```diff
@@ -641,9 +640,43 @@ void dbus_manager::disconnect_from_signals()
     stop_thread();
 }
 
-void dbus_manager::add_callback(CBENUM type, callback_func func)
+bool dbus_manager::dbus_list_name_to_owner()
 {
-    m_callbacks[type] = func;
+    DBusError error;
+
+    std::vector<std::string> names;
+    std::string owner;
+
+    DBusMessage * dbus_reply = nullptr;
+    DBusMessage * dbus_msg = nullptr;
+
+    // dbus-send --session --dest=org.freedesktop.DBus --type=method_call --print-reply /org/freedesktop/DBus org.freedesktop.DBus.GetNameOwner string:"org.mpris.MediaPlayer2.spotify"
+    if (nullptr == (dbus_msg = m_dbus_ldr.message_new_method_call("org.freedesktop.DBus", "/org/freedesktop/DBus", "org.freedesktop.DBus", "ListNames"))) {
+        std::cerr << "MANGOHUD: " << __func__ << ": unable to allocate memory for dbus message\n";
+        return false;
+    }
+
+    m_dbus_ldr.error_init(&error);
+    if (nullptr == (dbus_reply = m_dbus_ldr.connection_send_with_reply_and_block(dbus_mgr.get_conn(), dbus_msg, DBUS_TIMEOUT, &error))) {
+        m_dbus_ldr.message_unref(dbus_msg);
+        std::cerr << "MANGOHUD: " << __func__ << ": "<< error.message << "\n";
+        m_dbus_ldr.error_free(&error);
+        return false;
+    }
+
+    auto iter = DBusMessageIter_wrap(dbus_reply, &m_dbus_ldr);
+    if(not iter.is_array()) return false;
```

Comment:
Leaking dbus_msg etc, I think

Reply:
You're right. I'm currently looking into writing a (much smaller) wrapper for DBusMessage that cleans up after itself, but for the time being, I'll fix this.
Good documentation on the low-level C API is surprisingly hard to find. I'll have to figure out which messages need to be unref'd and which don't.